src/analyze/analyze-verify.c

```diff
@@ -52,21 +52,27 @@ static int prepare_filename(const char *filename, char **ret) {
         return 0;
 }
 
-static int generate_path(char **var, char **filenames) {
+static int generate_path(char **var, char **filenames, char *arg_root) {
         const char *old;
         char **filename;
 
         _cleanup_strv_free_ char **ans = NULL;
         int r;
 
         STRV_FOREACH(filename, filenames) {
-                char *t;
+                _cleanup_free_ char *t = NULL;
+                char *path_temp;
 
                 t = dirname_malloc(*filename);
                 if (!t)
                         return -ENOMEM;
 
-                r = strv_consume(&ans, t);
+                /* this allows verification of files from discrete images and directories */
+                path_temp = path_join(arg_root, t);
+                if (!path_temp)
+                        return -ENOMEM;
+
+                r = strv_consume(&ans, path_temp);
```

Comment:
this doesn't look right. if i run "systemd-anylze verify --root=/some/where foo.service then this will search for /some/where/foo.service only, but it should search for /some/where/usr/lib/systemd/system/foo.service and similar dirs.

support for the "root" logic needs to be added to the `Manager` object to some degree I guess.

Reply:
it does work when passing full paths to units though (relative to root/image), isn't that how this is typically used?

```
$ cat /tmp/img/usr/lib/systemd/system/test.service 
[Service]
ExecStart=echo hello
foo=bar
$ build/systemd-analyze verify --root /tmp/img /usr/lib/systemd/system/test.service
libbpf: load bpf program failed: Operation not permitted
libbpf: failed to load program 'sd_bind4'
libbpf: failed to load object 'socket_bind_bpf'
libbpf: failed to load BPF skeleton 'socket_bind_bpf': -1
/lib/systemd/system/plymouth-start.service:16: Unit configured to use KillMode=none. This is unsafe, as it disables systemd's process lifecycle management for the service. Please update your service to use a safer KillMode=, such as 'mixed' or 'control-group'. Support for KillMode=none is deprecated and will eventually be removed.
/etc/systemd/system/minissdpd.service.d/override.conf:4: Unknown key name 'StartLimitIntervalSec' in section 'Service', ignoring.
/tmp/img/usr/lib/systemd/system/test.service:3: Unknown key name 'foo' in section 'Service', ignoring.
```

Reply:
from the manpage, in the example for the `verify` verb:

```
           $ cat ./user.slice
           [Unit]
           WhatIsThis=11
           Documentation=man:nosuchfile(1)
           Requires=different.service

           [Service]
           Description=x

           $ systemd-analyze verify ./user.slice
```

Reply:
(That said, being able to handle root/image in the manager object is next on the TODO list, as it's required to easily check the security features/verb, so it will come too. But this looked nice and self-contained enough to be done immediately, to me)